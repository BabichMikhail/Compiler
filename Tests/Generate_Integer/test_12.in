begin 
	writeln('hello world!');
end.
