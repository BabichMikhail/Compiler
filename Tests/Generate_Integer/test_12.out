extern _printf
section .data
	base_str : times 256 db 0
	fmt : times 128 db 0
section .text
global _main
_main:
mov eax, 'hell'
mov [base_str + 0], eax
mov eax, 'o wo'
mov [base_str + 4], eax
mov eax, 'rld!'
mov [base_str + 8], eax
mov eax, 0x0
mov [base_str + 12], eax
push base_str
mov eax, '%s'
mov [fmt + 0], eax
mov eax, 0xA
mov [fmt + 2], eax
mov eax, 0x0
mov [fmt + 3], eax
push fmt
call _printf
add esp, 8
ret